fix(client): fluent api nulls

millsp opened the pull request:
Fixes https://github.com/prisma/prisma/issues/10687
Fixes https://github.com/prisma/prisma/issues/9138
Fixes https://github.com/prisma/prisma/issues/10656

danstarns reviewed (commented):
Is there a way we could add a regression test for this? Maybe something in the functional suite?

millsp committed:
fix(client): null on single result chaining

millsp commented:
> Is there a way we could add a regression test for this? Maybe something in the functional suite?

Thanks for the suggestion, I have added them and fixed some cases I've missed along the way.

millsp committed:
remove ts-toolbelt from checks

millsp committed:
also fix cases with rejectOnNotFound

millsp committed:
cleanup

millsp commented:
I'm now doing type assertions in the functional test setup thanks to your suggestion Dan.

millsp commented on packages/client/tests/functional/fluent-api-null/tests.ts:
```diff
@@ -0,0 +1,241 @@
+    const result = await prisma.resource.findFirst().children()
+
+    expect(result).toBeNull()
+    ;() => Test.checks([Test.check<U.Has<typeof result, null>, 1, Test.Pass>()])
```
Nice, I'll add that

millsp committed:
apply review comments
